code/modules/overmap/missions/acquire_mission.dm

```diff
@@ -0,0 +1,220 @@
+
+/datum/mission/acquire/turn_in()
+	del_container()
+	return ..()
+
+/datum/mission/acquire/give_up()
+	del_container()
+	return ..()
+
+/datum/mission/acquire/proc/current_num()
+	if(!container)
+		return 0
+	var/num = 0
+	for(var/A in container.contents)
+		num += atom_effective_count(A)
+		if(num >= num_wanted)
+			return num
+	return num
+
+/datum/mission/acquire/proc/atom_effective_count(atom/movable/A)
+	if(allow_subtypes ? !istype(A, objective_type) : A.type != objective_type)
+		return 0
+	if(count_stacks && istype(A, /obj/item/stack))
+		var/obj/item/stack/S = A
+		return S.amount
+	return 1
+
+/datum/mission/acquire/proc/del_container()
+	var/turf/cont_loc = get_turf(container)
+	for(var/atom/movable/A in container.contents)
+		if(atom_effective_count(A))
+			qdel(A)
+		else
+			A.forceMove(cont_loc)
+	recall_bound(container)
+	return
+
+/*
+	Acquire: True Love
+*/
+
+/datum/mission/acquire/true_love
+	name = "Diamond needed (urgent!!)"
+	weight = 3
+	value = 700
+	duration = 20 MINUTES
+	dur_mod_range = 0.2
+	container_type = /obj/item/storage/box/true_love
+	objective_type = /obj/item/stack/sheet/mineral/diamond
+	num_wanted = 1
+
+/datum/mission/acquire/true_love/New(...)
+	var/datum/species/beloved = pick(subtypesof(/datum/species))
+
+	desc = "I was going to gift \a [initial(objective_type.name)] to my [pick(list("beautiful", "handsome", "lovely"))] \
+			[initial(beloved.name)] [pick(list("boyfriend", "girlfriend", "lover", "SO", "spouse", "husband", "wife", "boywife"))], \
+			but I just lost it! Could you please find me a replacement? I don't have long!"
+	. = ..()
+
+/datum/mission/acquire/true_love/puce
+	name = "Puce crystal needed (urgent!!)"
+	weight = 1
+	objective_type = /obj/item/reagent_containers/food/snacks/grown/ash_flora/whitesands/puce
+
+/datum/mission/acquire/true_love/fireblossom
+	name = "Fireblossom needed (urgent!!)"
+	weight = 1
+	objective_type = /obj/item/reagent_containers/food/snacks/grown/ash_flora/whitesands/fireblossom
+
+/datum/mission/acquire/true_love/icepepper
+	name = "Icepepper needed (urgent!!)"
+	weight = 1
+	objective_type = /obj/item/reagent_containers/food/snacks/grown/icepepper
+
+/datum/mission/acquire/true_love/strange_crystal
+	name = "Strange crystal needed (urgent!!!)"
+	value = 1000
+	weight = 1
+	objective_type = /obj/item/strange_crystal
+
+/*
+		Acquire: The Creature
+*/
+
+/datum/mission/acquire/creature
+	name = "Capture a goliath"
+	desc = "I require a live goliath for research purposes. Trap one within the given \
+			Lifeform Containment Unit and return it to me and you will be paid handsomely."
+	value = 1500
+	duration = 30 MINUTES
+	weight = 6
+	container_type = /obj/structure/closet/mob_capture
+	objective_type = /mob/living/simple_animal/hostile/asteroid/goliath
+	num_wanted = 1
+	count_stacks = FALSE
+
+/datum/mission/acquire/creature/atom_effective_count(atom/movable/A)
+	. = ..()
+	if(!.)
+		return
+	var/mob/M = A
+	if(M.stat == DEAD)
+		return FALSE
```

Comment:
you can do return (M.stat != DEAD) here

Reply:
i think that's a bit less clear -- and also this should probably actually be "return 0", not "return FALSE"